ios/pod_helpers.rb

```diff
@@ -46,8 +46,8 @@ def find_file(file_name, current_dir)
 
 def new_architecture_enabled?(options, react_native_version)
   supports_new_architecture = supports_new_architecture?(react_native_version)
-  supports_new_architecture && ENV.fetch('RCT_NEW_ARCH_ENABLED',
-                                         options[:fabric_enabled] || options[:turbomodule_enabled])
+  enabled = ENV.fetch('RCT_NEW_ARCH_ENABLED', options[:fabric_enabled] || options[:turbomodule_enabled] ? '1' : '0') == '1'
+  supports_new_architecture && enabled
```

Comment:
Let's return early so the rest becomes a little easier to read:
```suggestion
  return false unless supports_new_architecture?(react_native_version)

  enabled = options[:fabric_enabled] || options[:turbomodule_enabled]
  ENV.fetch('RCT_NEW_ARCH_ENABLED', enabled ? '1' : '0') != '0'
```

Reply:
I incorporated this and also renamed the local variable `enabled` to `dependent_options_enabled` to convey the semantics better.